Guard length of url for bz reporting

rvykydal opened the pull request:
Low priority, the first commit can be perhaps merged separately.
https://issues.redhat.com/browse/INSTALLER-3612

Example of truncated comment:
(The message should appear really exceptionally - we don't expect automatically generated Detail/comment to exceed the 7000 characters. It is there so it is clear what really happened (the comment was truncated) in such a rare case.)

![Screenshot from 2023-08-22 14-34-38](https://github.com/rhinstaller/anaconda/assets/1220237/12195c73-fd8f-42d1-98f8-4427300b7b2d)

rvykydal commented:
/kickstart-test --waive webui-only

KKoukiou commented on ui/webui/src/components/Error.jsx:
```diff
@@ -53,6 +53,21 @@ export const bugzillaPrefiledReportURL = (productQueryData) => {
+    const newUrl = new URL(reportURL);
+    const searchParamsLimits = [
+        { param: "short_desc", length: 256, message: null },
+        { param: "comment", length: 7800, message: "\nTHE COMMENT EXCEEDING THE LENGTH LIMIT WAS TRUNCATED" },
```
Should not there be an error in the dialog form instead of filling this warning in the URL?

rvykydal replied:
The comment (pre-filed Detail) is generated by Anaconda, there is no user input at this point (currently it is used only in CriticalError where it is the exception content) so I don't think it makes sense to inform user about it. It is meant as an information for developers, not for the user. So maybe we should remove it completely (and just log it) so it does not confuse user in the BZ report form.

VladimirSlavik replied:
I just realized, there is no mention of what's the actual limit? What is 7800?

rvykydal replied:
> I just realized, there is no mention of what's the actual limit? What is 7800?

I'll add a comment in the code.

VladimirSlavik commented:
/webui-test

VladimirSlavik commented on ui/webui/src/components/Error.jsx:
```diff
@@ -83,6 +98,7 @@ export const BZReportModal = ({
 
     const openBZIssue = (reportURL, logFile, logContent) => {
+        reportURL = ensureMaximumReportURLLength(reportURL);
         reportURL = addLogAttachmentCommentToReportURL(reportURL, logFile);
```
Is this the right order? First you trim it and then add the comment.

rvykydal replied:
Yes it is. There number is taking the added comment (of ~ static length) into account. We want the added comment to be there always, irrespective of the message being cut. I'll explain it in a comment in the code as well.

rvykydal commented:
So I added some comments in the code and removed the message informing about the Detail being cut from the Detail. The message asking to attach the log is still added after the cut detail.

![Screenshot from 2023-09-05 15-29-40](https://github.com/rhinstaller/anaconda/assets/1220237/7b60fbbd-f3ab-490f-9232-c16d2c3f8fb7)

VladimirSlavik reviewed (approved):
Thanks!

rvykydal commented:
/kickstart-test --waive webui-only

VladimirSlavik commented:
/webui-test

rvykydal commented:
/kickstart-test --waive webui-only
